### PR TITLE
ga-migration script part II

### PR DESCRIPTION
**Pull request by mvidalgarcia**

### CODE REVIEW

File: automation_tools/scripts/ga-migration/templates/services/tests.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+          # You can see the complete list of services and versions that are available at:
+          # https://docker-services-cli.readthedocs.io/en/latest/configuration.html
+          python-version: [3.6, 3.7, 3.8, 3.9]
+          requirements-level: [min, pypi]
+          {{ "cache-service: [memcached, redis]" if cache }}
+          {{ "db-service: [postgresql9, postgresql11, mysql5, mysql8]" if db }}
+          {{ "mq-service: [rabbitmq, redis]" if mq }}
+          {{ "search-service: [elasticsearch6, elasticsearch7]" if search }}
+
+          exclude:
```
Comment (diegodelemos) [inline automation_tools/scripts/ga-migration/templates/services/tests.yml:37]: Major: we should remember to come up with the list mentioned in https://github.com/inveniosoftware/cookiecutter-invenio-module/pull/141#discussion_r534029729, to we have a list of excluded builds by default so when we make all the PRs we do not consume our quota.

File: automation_tools/scripts/ga-migration/utils.py [hunk 2]
```diff
@@ -5,13 +5,75 @@
...
+
+
+def get_repo_services(repo_name):
+    """Get repository services."""
+    travis_raw_url = f"https://raw.githubusercontent.com/inveniosoftware/{repo_name}/master/.travis.yml"
+    is_migrated = not requests.get(travis_raw_url).ok
+
+    search_terms = {
+        'postgres': 'db',
+        '- redis': 'cache',
```
Comment (diegodelemos) [inline automation_tools/scripts/ga-migration/utils.py]: Why not `redis-service`? e.g. https://github.com/inveniosoftware/invenio-cache/blob/50879bbf441718eebb81a0356852b76590166824/.travis.yml#L24

Comment (mvidalgarcia) [reply automation_tools/scripts/ga-migration/utils.py]: I found **only** `- redis` in other repos like [invenio-indexer](https://github.com/inveniosoftware/invenio-indexer/blob/0b50ed3f8a3ec683ae6a1e697cd04222a0d8c569/.travis.yml#L25) so in the end I think lowest common denominator is that. What do you think?

Comment (diegodelemos) [reply automation_tools/scripts/ga-migration/utils.py]: Good enough, thanks!

**Comment by diegodelemos** [comment]
With:
```console
automation-tools $ git log -1 --oneline
18915be (HEAD -> ga-migration, upstream/ga-migration) ga-migration: add logic to build from templates
invenio-stats $ git log -1 --oneline
f73609b (HEAD -> master, tag: v1.0.0a18, upstream/master, origin/master, origin/HEAD) release: v1.0.0a18
```
I get the following error:
```console
$ python main.py --targetpath ~/project/invenio/src/invenio-stats # also tried on invenio-queues
INFO:root:TASK: Building and copying tests.yml template...
Traceback (most recent call last):
  File "main.py", line 141, in <module>
    pipeline()
  File "/Users/rodrigdi/.pyenv/versions/gh-migrate-helper/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/rodrigdi/.pyenv/versions/gh-migrate-helper/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/rodrigdi/.pyenv/versions/gh-migrate-helper/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/rodrigdi/.pyenv/versions/gh-migrate-helper/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "main.py", line 134, in pipeline
    migrate_repo(targetpath)
  File "main.py", line 71, in migrate_repo
    build_template(
  File "/Users/rodrigdi/project/invenio/src/automation-tools/automation_tools/scripts/ga-migration/utils.py", line 63, in build_template
    content = render_template(
  File "/Users/rodrigdi/project/invenio/src/automation-tools/automation_tools/scripts/ga-migration/utils.py", line 73, in render_template
    template = JinjaEnv.get_template(template_file)
  File "/Users/rodrigdi/.pyenv/versions/gh-migrate-helper/lib/python3.8/site-packages/jinja2/environment.py", line 883, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/Users/rodrigdi/.pyenv/versions/gh-migrate-helper/lib/python3.8/site-packages/jinja2/environment.py", line 857, in _load_template
    template = self.loader.load(self, name, globals)
  File "/Users/rodrigdi/.pyenv/versions/gh-migrate-helper/lib/python3.8/site-packages/jinja2/loaders.py", line 115, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "/Users/rodrigdi/.pyenv/versions/gh-migrate-helper/lib/python3.8/site-packages/jinja2/loaders.py", line 197, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: services/tests.yml
```

**Comment by mvidalgarcia** [comment]
@diegodelemos It should be fixed by [this change](https://github.com/inveniosoftware/automation-tools/pull/9/commits/f3d5c427e4da6def6058768c628525c712746475#diff-334e9270f835f084ff3ae3f38cb1c0274ccf1f2301a480a3afc47481f260a30fR19), thanks for reporting 👍
Also, already migrated repos should be taken into account now.

File: automation_tools/scripts/ga-migration/templates/services/run-tests.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+
+#!/usr/bin/env bash
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or modify
```
Comment (diegodelemos) [inline automation_tools/scripts/ga-migration/templates/services/run-tests.sh]: Could we change this to the generic Invenio license header? e.g. https://github.com/inveniosoftware/automation-tools/pull/9/files#diff-707cee7a3f79b3ef314738bb5c26a4fb63dc739a84046a96c0eaeed37374a374

Comment (mvidalgarcia) [reply automation_tools/scripts/ga-migration/templates/services/run-tests.sh]: Yes, I have it locally, pushing soon.

Comment (mvidalgarcia) [reply automation_tools/scripts/ga-migration/templates/services/run-tests.sh]: Pushed with some other improvements:

```
- Adapt to latest `docker-services-cli` version
- Fix various issues with Jinja white-spaces
- Emulate join function to generate EXTRAS
```

File: automation_tools/scripts/ga-migration/README.md [hunk 4]
```diff
@@ -8,12 +8,34 @@ $ source ~/.virtualenvs/ga-migration/bin/activate
...
+```bash
+# e.g.
+# git clone https://github.com/inveniosoftware/invenio-charts-js
+python main.py --targetpath=invenio-charts-js
+```
+
+## Full pipeline
 
-# Migrate all repos on config.py
-$ python main.py
```
Comment (mvidalgarcia) [inline automation_tools/scripts/ga-migration/README.md:18]: Why have you deleted this option? It's useful to bulk-migrate repos

**Review by lnielsen — COMMENTED** [review]
Comment: Just noticed the commit history - please try to keep logical commits instead of many smaller commits. It get's very difficult both to review, and go back and find bugs when there's a lot of small commits.

File: automation_tools/scripts/ga-migration/README.md (hunk 4 above)
Comment (avivace) [reply automation_tools/scripts/ga-migration/README.md:18]: Woops

**Review by diegodelemos — APPROVED** [review]

**Pull request merged**